src/utils/swapAndXcallParams.ts

```diff
@@ -0,0 +1,26 @@
+import { ethers } from 'ethers';
+
+export const getSwapAndXcallParams = (
+  originDomain: string,
+  destinationDomain: string,
+  fromAsset: string,
+  toAsset: string,
+  amountIn: string,
+  to: string, // mean target address.
+  callData: string,
+  relayerFeeInNativeAsset: string
+) => {
+  const fromAssetOrigin =
+    fromAsset === '0xeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee' ? ethers.constants.AddressZero : fromAsset;
```

Comment:
you can use the `PROTOCOL_TOKEN_ADDRESS` constant here

Reply:
Noted ser ✅